### ISSUE TITLE
TFVC plugin just seems to contain random stuff

### ISSUE THREAD

**Issue by AndyWDev** [microsoft/azure-repos-vscode#466]
This issue was originally raised using the VS Code reporting feature. But that reports the issue in the wrong place.

I am trying to use the TFVC plugin for TFS version control. I wish to check in two new files for a work item. I am using on premisis TFS.

When I view the "Included Files" menu it contains hundreds of files that have been checked out by other developers". This information is useless to me and confusing and not relevent to the work being done (only two files are invovled).

There are also various icons next to each one (E, A, D) but I cant find any documentation saying what they are).

When trying to check in the two files, the plugin output gives the impression that it is trying to process the hundreds of files and eventually fails with "Execution of the TFVC command line failed unexpectedly".

I was also unable to get it to work by specifying a TFS work-item number to check code in against.

If I use source code explorer in Visual Studio, this issue does not occur.

**Comment by Schpeyeder** [comment]
Andy,  Can you share with me how you managed to get this to connect to your local TFS instance in prem?  I have several open requests / reports that this doesn’t seem to work with no reply from the dev team.  I’ve tried lots of variations and it seems as though auth is just failing for what ever reason.  One prospect was that the node module was corrupt they bundled in and it was recommended that I recompile the source with a patched version of the auth module.  No thanks.

Anyway, I’d be interested in knowing your setup / confit to at LEAST get it to connect successfully.  Appreciate any help you may be able to throw my way, and I apologize for answering your question with my own question.

**Comment by AndyWDev** [comment]
Hi,
 I dont think I can be of much use, as I have just started using the TF.exe command line tool from a Visual Studio command prompt (or a VS Code Terminal Window) or from inside Visual Studio 2017 (Team Explorer window).   Using VS Code with version control has just proved too difficult and time consuming for me at the moment.

When I installed it originally, VS code required manually editing a settings file directly - I seem to remember there was a place I needed to add my TFS server address and a couple of extra settings to configure - if I could find that settings file I could tell you what I did..  I am presently stuck not being able to change team projects.

Under the new version of VS Code, the plugin no longer shows up in the extensions manager and I have not been able to find where the TFS server is configured.  In the settings.json (the documentation appears incorrect in the location of this)  file there is a tfvc.location property which just points to the TF.exe in my visual studio directory.

The other issue that I have is with the concept of the workspace in VS Code. I still have not been able to work this out, as there appears to be a workspace in VS Code and here is also a Workspace in TFS (where you check out code to) and the documentation isnt very clear on which one does what.  Visual Studio has a configure workspace menu but VS Code does not seem to conform to this.

In visual studio team explorer there is an option on the check-in page under excluded changes, there is a 'view options' menu, which gives the ability to view files relevant to the solution or across TFS  (Show All, Show Solution Changes) - VS Code does not appear to support this and does a show all - which is what my original post was for (it needs to be show solution by default I think).
